scripts/sop/README.md

```diff
@@ -0,0 +1,30 @@
+# Downloading Wikipedia Corpus
+
+We use the preprocessing code from https://github.com/NVIDIA/DeepLearningExamples/tree/master/PyTorch/LanguageModeling/BERT#getting-the-data
+and the bash scripts provided here is used to help with streamlining the data generation in the NVIDIA repository.
+
+First, git clone https://github.com/NVIDIA/DeepLearningExamples.git.
+Then, move create_wiki_data.sh and get_small_english_wiki.sh into DeepLearningExamples/PyTorch/LanguageModeling/BERT/data.
+
+Then, follow the instructions below:
+
+NVIDIA script download the latest Wikipedia dump. We use the Wikipedia dump 2020-03-01.
+To download the Wikipedia dump 2020-03-01, replace line 29 of `DeepLearningExamples/PyTorch/LanguageModeling/BERT/data/WikiDownloader.py`:
+`'en' : 'https://dumps.wikimedia.org/enwiki/latest/enwiki-latest-pages-articles.xml.bz2',` with `'en' : `https://dumps.wikimedia.org/enwiki/20200301/enwiki-20200301-pages-articles.xml.bz2`.  
+
+The data creation for SOP is almost the same as MLM, except you need to edit the following.
+In `DeepLearningExamples/PyTorch/LanguageModeling/BERT/data/TextSharding.py`, replace line 55:
+`self.articles[global_article_count] = line.rstrip()` with `self.articles[global_article_count] = line.rstrip() + "\n ========THIS IS THE END OF ARTICLE.========"`.
+This is because SOP requires a signal for the end of each Wikipedia article.
+
+Run `bash create_wiki_sop_data.sh $lang $save_directory`
+The NVIDIA code supports English (en) and Chinese (zh) wikipedia.
+
+For example, to download and process English Wikipedia and save it in `~/Download` directory, run
+`bash create_wiki_sop_data.sh en ~/Download`
+
```

Comment:
Heads up — there are some steps that I needed to take that I don't think are documented here. Here are my notes from my debugging process (they're rough but maybe they'll help with updating these instructions):

> `create_wiki_sop_data.sh` takes cmd line args, `create_wiki_sop_data.sh` exports the second arg as `BERT_PREP_WORKING_DIR`, `create_wiki_sop_data.sh` checks if dir `wikiextractor` exists and if not clones it. Later, `create_wiki_sop_data.sh` calls `bertPrep.py`, and `bertPrep.py` (called with default `args.skip_wikiextractor == 0`) expects `WikiExtractor.py` to be at `/workspace/wikiextractor/WikiExtractor.py`, but `create_wiki_sop_data.sh` didn't put it under `/workspace/`.

So, after that I modified `bertPrep.py` at L80 to point to where `create_wiki_sop_data.sh` cloned `wikiextractor`.

Reply:
Included.